Ruins/src/main/java/atomicstryker/ruins/common/RuinTemplateRule.java

```diff
@@ -606,7 +606,7 @@ else if (dataString.startsWith("HardChest"))
         else if (dataString.startsWith("ChestGenHook:"))
         {
             String[] s = dataString.split(":");
-            int targetCount = s.length > 2 ? Integer.valueOf(s[s.length-2].split("-")[0]) : 0;
+            int targetCount = s.length > 2 ? Integer.valueOf(s[s.length-1].split("-")[0]) : 0;
             addChestGenChest(world, random, x, y, z, s[1], targetCount, blocknum, rotate);

```

Comment:
You're only passing the namespace portion of the loot table resource location, s[1], to addChestGenChest(). So, for example, "foo:chests/bar" will be interpreted as "minecraft:foo"...not what you want.

A simpler solution might be to leave the original code as-is, but just massage the s array a bit before using it, glomming the pieces of the loot table location together:

```suggestion
            if (s.length > 3)
            {
                s[1] = String.join(":", s[1], s[2]);
                s[2] = s[3];
            }
```